File: cdm-sources/cdm/observable/asset/calculatedrate/functions/IndexValueObservation.java
package cdm.observable.asset.calculatedrate.functions;

import cdm.observable.asset.InterestRateIndex;
import com.google.inject.ImplementedBy;
import com.rosetta.model.lib.functions.RosettaFunction;
import com.rosetta.model.lib.records.Date;
import java.math.BigDecimal;


@ImplementedBy(IndexValueObservation.IndexValueObservationDefault.class)
public abstract class IndexValueObservation implements RosettaFunction {

	/**
	* @param observationDate 
	* @param interestRateIndex 
	* @return observedValue 
	*/
	public BigDecimal evaluate(Date observationDate, InterestRateIndex interestRateIndex) {
		BigDecimal observedValue = doEvaluate(observationDate, interestRateIndex);
		
		return observedValue;
	}

	protected abstract BigDecimal doEvaluate(Date observationDate, InterestRateIndex interestRateIndex);

	public static class IndexValueObservationDefault extends IndexValueObservation {
		@Override
		protected BigDecimal doEvaluate(Date observationDate, InterestRateIndex interestRateIndex) {
			BigDecimal observedValue = null;
			return assignOutput(observedValue, observationDate, interestRateIndex);
		}
		
		protected BigDecimal assignOutput(BigDecimal observedValue, Date observationDate, InterestRateIndex interestRateIndex) {
			return observedValue;
		}
	}
}

File: cdm-sources/src/cdm/observable/asset/calculatedrate/functions/IndexValueObservation.py
# pylint: disable=line-too-long, invalid-name, missing-function-docstring, missing-module-docstring, superfluous-parens
# pylint: disable=wrong-import-position, unused-import, unused-wildcard-import, wildcard-import, wrong-import-order, missing-class-docstring
from __future__ import annotations
import sys
import datetime
import inspect
from decimal import Decimal
from rosetta.runtime.utils import *
from rosetta.runtime.func_proxy import replaceable, create_module_attr_guardian
from cdm.observable.asset.InterestRateIndex import InterestRateIndex

__all__ = ['IndexValueObservation']


@replaceable
def IndexValueObservation(observationDate: datetime.date, interestRateIndex: InterestRateIndex) -> Decimal:
    """
    Retrieve the values of the supplied index on the specified observation date.
    
    Parameters 
    ----------
    observationDate : date
    
    interestRateIndex : InterestRateIndex
    
    Returns
    -------
    observedValue : number
    
    """
    self = inspect.currentframe()
    
    
    observedValue = rosetta_resolve_attr(self, "observedValue")
    
    
    return observedValue

sys.modules[__name__].__class__ = create_module_attr_guardian(sys.modules[__name__].__class__)
